File: 백준/Silver/2559. 수열/수열.py
import sys

input = sys.stdin.readline

INF = float("inf")


n, k = map(int, input().split())
temps = [0] + list(map(int, input().split()))
prefixSum = [0] * (n + 1)
for i in range(1, n + 1):
    prefixSum[i] = temps[i] + prefixSum[i - 1]

answer = -INF
for i in range(k, n + 1):
    answer = max(answer, prefixSum[i] - prefixSum[i - k])

print(answer)

File: 백준/Silver/2559. 수열/수열.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

  static StringBuilder sb = new StringBuilder();
  static int n, k;
  static int[] prefixSum;

  public static void main(String[] args) throws IOException {
    setUp();

    int answer = Integer.MIN_VALUE;
    for (int end = k; end < n + 1; end++) {
      answer = Math.max(answer, prefixSum[end] - prefixSum[end - k]);
    }
    sb.append(answer);

    output();
  }

  private static void setUp() throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());
    n = Integer.parseInt(st.nextToken());
    k = Integer.parseInt(st.nextToken());
    prefixSum = new int[n + 1];
    st = new StringTokenizer(br.readLine());
    for (int i = 1; i < n + 1; i++) {
      prefixSum[i] = prefixSum[i - 1] + Integer.parseInt(st.nextToken());
    }
  }

  private static void output() throws IOException {
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }

}